h2o-algos/src/main/java/hex/tree/BuildHistogram.java

```diff
@@ -57,25 +57,76 @@
  *    exp(nthreads-pre-column) = max(1,H2O.NUMCPUS - num_cols)
  *
  */
-public class ScoreBuildHistogram2 extends ScoreBuildHistogram {
-  transient int []   _cids;
+public class BuildHistogram extends MRTask<BuildHistogram> {
```

Comment:
Why did you decide to remove 'Score' from the name? It still scores histogram, or not?

Reply:
yes it does but it is not `scoring` in the sense that is used elsewehere, I think the `BuildHistogram` name is just fine

Reply:
yes, but this class is full of "scoring" terminology in javadoc - imho it correctly reflects the 2 phase approach - it does more than just to build the histograms, it is not an idempotent action

Reply:
adding back Score to the name